include/picolibrary/testing/unit/adc.h

```diff
@@ -0,0 +1,204 @@
+ * \brief Analog-to-Digital Converter (ADC) unit testing facilities.
+ */
+namespace picolibrary::Testing::Unit::ADC {
+
+/**
+ * \brief Mock blocking, single sample ADC.
+ *
+ * \tparam Sample_Type The integral type used to hold a sample.
+ */
+template<typename Sample_Type>
+class Mock_Blocking_Single_Sample_Converter {
+  public:
+    /**
+     * \copydoc picolibrary::ADC::Blocking_Single_Sample_Converter_Concept::Sample
+     */
+    using Sample = Sample_Type;
+
+    /**
+     * \brief Movable mock blocking, single sample ADC handle.
+     */
+    class Handle {
+      public:
+        /**
+         * \copydoc picolibrary::ADC::Blocking_Single_Sample_Converter_Concept::Sample
+         */
+        using Sample = Sample_Type;
+
+        /**
+         * \brief Constructor.
+         */
+        Handle() noexcept = default;
+
+        /**
+         * \brief Constructor.
+         *
+         * \param[in] mock_blocking_single_sample_converter The mock blocking, single
+         *            sample ADC.
+         */
+        Handle( Mock_Blocking_Single_Sample_Converter & mock_blocking_single_sample_converter ) noexcept :
+            m_mock_blocking_single_sample_converter{ &mock_blocking_single_sample_converter }
+        {
+        }
+
+        /**
+         * \brief Constructor.
+         *
+         * \param[in] source The source of the move.
+         */
+        Handle( Handle && source ) noexcept = default;
+
+        /**
+         * \todo #29
+         */
+        Handle( Handle const & ) = delete;
+
+        /**
+         * \brief Destructor.
+         */
+        ~Handle() noexcept = default;
+
+        /**
+         * \brief Assignment operator.
+         *
+         * \param[in] expression The expression to be assigned.
+         *
+         * \return The assigned to object.
+         */
+        auto operator=( Handle && expression ) noexcept -> Handle & = default;
+
+        /**
+         * \todo #29
+         *
+         * \return
+         */
+        auto operator=( Handle const & ) = delete;
+
+        /**
+         * \brief Initialize the ADC's hardware.
+         *
+         * \return Nothing if initializing the ADC's hardware succeeded.
+         * \return An error code if initializing the ADC's hardware failed.
+         */
+        auto initialize()
+        {
+            return m_mock_blocking_single_sample_converter->initialize();
+        }
+
+        /**
+         * \copydoc picolibrary::ADC::Blocking_Single_Sample_Converter_Concept::min()
+         */
+        auto min() const noexcept
+        {
+            return m_mock_blocking_single_sample_converter->min();
+        }
+
+        /**
+         * \copydoc picolibrary::ADC::Blocking_Single_Sample_Converter_Concept::max()
+         */
+        auto max() const noexcept
+        {
+            return m_mock_blocking_single_sample_converter->max();
+        }
+
+        /**
+         * \copydoc picolibrary::ADC::Blocking_Single_Sample_Converter_Concept::sample()
```

Comment:
The concept's documentation covers cases where the operation can and cannot fail, and thus should not be used here.